### PR TITLE
Allow deletions of projects

### PR DESCRIPTION
**Pull request by henrik997**
This PR contains the functionality to stage and delete projects including associated user capellamodels, a overview of staged project for deletion to administrators and a makefile for the t4c-server mock.  
Only project managers or admins can stage projects for deletion and the project deletion must be approved by an admin.

This fixes issue #24.

Depends on the PR https://github.com/DSD-DBS/capella-collab-manager/pull/189

### CODE REVIEW

**Comment by MoritzWeber0** [comment]
We have to wait until #40 is implemented and merged before we can merge this PR. There will be some merge conflicts that will be easier to fix after the merge. 
In addition, cascading deletion of TeamForCapella repositories, Git repositories and Backup jobs have to be handled. This is also easier after #40 is merged.

File: frontend/src/app/settings/admin-settings/staged-to-delete-overview/staged-to-delete-overview.component.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { RepositoryService } from 'src/app/services/repository/repository.service';
```
[the remark this answers is not in the document]
Comment (henrik997) [reply frontend/src/app/settings/admin-settings/staged-to-delete-overview/staged-to-delete-overview.component.ts:2]: the whole component is a relict

File: frontend/src/app/settings/requests/staged-projects-overview/staged-projects-overview.component.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            for (let i = 0; i < this.stagedProjects.length; i++) {
+              if (this.stagedProjects[i].name == project_name) {
+                this.stagedProjects.splice(i, 1);
+                break;
+              }
+            }
+          },
+        });
+      }
+    });
```
[the remark this answers is not in the document]
Comment (henrik997) [reply frontend/src/app/settings/requests/staged-projects-overview/staged-projects-overview.component.ts:47]: In fact, I wanted to it this way you propose, but I had not enough time.

**Comment by MoritzWeber0** [comment]
Given the number of conflicts, it's easier to just start again with the current branch as base. Will close this issue.

**Pull request closed without merging**